src/objectprotocol.rs

```diff
@@ -176,15 +171,10 @@ pub trait ObjectProtocol {
     fn get_type_ptr(&self) -> *mut ffi::PyTypeObject;
 
     /// Gets the Python base object for this object.
-    fn get_base(&self) -> &<Self as PyTypeInfo>::BaseType
+    fn get_base<'py>(&'py self) -> &'py <Self as PyTypeInfo>::BaseType
     where
-        Self: PyTypeInfo;
-
-    /// Gets the Python base object for this object.
-
-    fn get_mut_base(&mut self) -> &mut <Self as PyTypeInfo>::BaseType
-    where
-        Self: PyTypeInfo;
+        Self: PyTypeInfo,
+        <Self as PyTypeInfo>::BaseType: FromPyPointer<'py>;
```

Comment:
Might be able to do `for <'py> FromPyPointer<'py>` instead of needing to add `'py` everywhere?

Reply:
Good catch, thanks :+1: